packages/yourloops/components/device/utils/device.utils.ts

```diff
@@ -125,37 +124,39 @@ export const sortPumpSettingsParameterByLevel = (historyParameters: ParametersCh
 }
 
 export const sortParameterList = (parameters: ParameterConfig[]): ParameterConfig[] => {
-  const aggressivenessParameters = parameters.filter(parameter => parameter.unit === Unit.Percent)
-  const insulinParameters = parameters.filter(parameter => parameter.unit === Unit.InsulinUnit || parameter.unit === Unit.InsulinUnitPerGram)
-  const mealParameters = parameters.filter(parameter => parameter.unit === Unit.Gram)
-  const weightParameters = parameters.filter(parameter => parameter.unit === Unit.Kilogram)
-  const thresholdParameters = parameters.filter(parameter => parameter.unit === Unit.MilligramPerDeciliter || parameter.unit === Unit.MmolPerLiter)
-  const minutesParameter = parameters.filter(parameter => parameter.unit === Unit.Minute)
+  const settingsOrder = [
+    'MEDIUM_MEAL_BREAKFAST',
+    'MEDIUM_MEAL_LUNCH',
+    'MEDIUM_MEAL_DINNER',
+    'TOTAL_INSULIN_FOR_24H',
+    'WEIGHT',
+    'PATIENT_GLY_HYPER_LIMIT',
+    'PATIENT_GLY_HYPO_LIMIT',
+    'PATIENT_GLYCEMIA_TARGET',
+    'PATIENT_BASAL_AGGRESSIVENESS_FACTOR_LEVEL_IN_EUGLYCAEMIA',
+    'BOLUS_AGGRESSIVENESS_FACTOR',
+    'MEAL_RATIO_BREAKFAST_FACTOR',
+    'MEAL_RATIO_LUNCH_FACTOR',
+    'MEAL_RATIO_DINNER_FACTOR',
+    'SMALL_MEAL_BREAKFAST',
+    'LARGE_MEAL_BREAKFAST',
+    'SMALL_MEAL_LUNCH',
+    'LARGE_MEAL_LUNCH',
+    'SMALL_MEAL_DINNER',
+    'LARGE_MEAL_DINNER'
+  ]
 
-  const sortedMealParameters = mealParameters.length > 0 ? sortMealParameters(mealParameters) : mealParameters
-  return [...insulinParameters, ...weightParameters, ...thresholdParameters, ...aggressivenessParameters, ...sortedMealParameters, ...minutesParameter]
-}
-
-const sortMealParameters = (parameters: ParameterConfig[]): ParameterConfig[] => {
-  const breakfastMeals = []
-  const lunchMeals = []
-  const dinnerMeals = []
-  parameters.forEach((parameter) => {
-    const splitParameterKey = parameter.name.split('_')
-    switch (splitParameterKey[2]) {
-      case DeviceMeals.Breakfast:
-        breakfastMeals.push(parameter)
-        break
-      case DeviceMeals.Lunch:
-        lunchMeals.push(parameter)
-        break
-      case DeviceMeals.Dinner:
-        dinnerMeals.push(parameter)
-        break
+  parameters.sort((a, b) => {
+    const aIdx = settingsOrder.indexOf(a.name)
```

Comment:
<!-- Nitpicks are trivial preference-based requests. These should be non-blocking by nature.  -->
**🤓 nitpick:** ‏`Index` rather than `Idx` to improve immediate readability would be nice 😄 ‏